auditlog/proxy_decorator_test.go

```diff
@@ -0,0 +1,142 @@
+package auditlog
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+
+	"github.com/ory/x/logrusx"
+)
+
+type MockProxy struct {
+	mock.Mock
+	t         *testing.T
+	roundTrip func(t *testing.T, r *http.Request) (*http.Response, error)
+	director  func(t *testing.T, r *http.Request)
+}
+
+func (m *MockProxy) RoundTrip(r *http.Request) (*http.Response, error) {
```

Comment:
Ok, it seems I didn't get the idea of the test :) because you kind of had to create your own mock implementation. But the comment can still be helpful as an informational comment.

Reply:
Thanks for the advice. Mockery is great :) I will definitely use it in my next tasks.

Yep, I had to create my mock implementations here. I think the `MockSender` is too tricky to being generated automatically.